src/server/pkg/storage/chunk/writer.go

```diff
@@ -92,6 +92,7 @@ type worker struct {
 	prev                 *prevChanSet
 	next                 *nextChanSet
 	stats                *stats
+	noUpload             bool
```

Comment:
What's the purpose of `noUpload`? It seems like a `Writer` wouldn't be of much use if it didn't upload, but I assume I'm misunderstanding what this does in some way.

Reply:
`noUpload` is used when we just want the chunk storage layer to give us the metadata for data that is copied into it. This is used for computing the hashes of files without uploading the content. From the PR comment above:
> One merge read is exclusively for computing file hashes, and does a no upload copy (does not upload to object storage) to a dummy fileset, which will return the resolved index entries for the files (which contain the hashes)

Reply:
Makes sense, thanks for clarifying.